modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -710,7 +710,7 @@ public void postProcessParameter(CodegenParameter parameter) {
     //override with any special handling of the entire OpenAPI spec document
     @SuppressWarnings("unused")
     public void preprocessOpenAPI(OpenAPI openAPI) {
-        if (useOneOfInterfaces) {
+        if (useOneOfInterfaces && openAPI.getComponents() != null) {
```

Comment:
👍

Reply:
The problem is NPE in 2 testcases in SpringCodegenTest, testAdditionalPropertiesPutForConfigValues and
testInitialConfigValues.
I don't know if it can happen in real execution, because of it i used this check.